modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,158 @@
+= Decode and Encode Base64
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, transform, transformer, format, xml, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping
+
+The DataWeave examples show how to convert a file stream into Base64 and to convert a Base64 string into a file stream. They use a PDF file as the input and output.
+//LINK TO DW 1.0 EXAMPLES:
+include::partial$dataweave1-links.adoc[tag=dataweave1Examples]
+
+[[xml_config]]
+== XML Configuration Example
+
+The configuration XML in the following procedure is a Mule flow that encodes and decodes a PDF file. It reads the file from the specified local directory using the Read operation from the File module. It writes the PDF to the specified directory.
+
+You can create this example in Anypoint Studio:
+
+. Create a Mule project in Anypoint Studio.
+. Use *Add Module* in the Mule Palette to add the File module.
+. Copy and paste the following example between the `<mule/>` elements of a Mule flow in Anypoint Studio.
++
+[source,XML,linenums]
+----
+<http:listener-config name="HTTP_Listener_config"
+                      doc:name="HTTP Listener config" >
+    <http:listener-connection
+                            host="0.0.0.0"
+                            port="8081" />
+</http:listener-config>
+<file:config name="File_Config"
+             doc:name="File Config" >
+    <file:connection workingDir="/Users/me/Downloads" />
+</file:config>
+<flow name="encode-decode-base64">
+    <http:listener doc:name="Listener"
+                 path="/transform"
+                 config-ref="HTTP_Listener_config"/>
+    <file:read doc:name="Read"
+             config-ref="File_Config"
+             path="pdf-test.pdf" />
+    <ee:transform doc:name="Transform Message">
+        <ee:message>
+          <ee:set-payload><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output text/plain
+---
+toBase64(payload as Binary)]]></ee:set-payload>
+        </ee:message>
+    </ee:transform>
+    <ee:transform doc:name="Transform Message" >
+        <ee:message >
+          <ee:set-payload ><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output application/octet-stream
```

Comment:
If these are docs for 4.3 exclusively, we could use `output binary` which is more concise and easier to remember, with a note perhaps saying its equivalent to writing that mime type.

Reply:
Actually, I think this would be the correct use for custom mime type using `output binary with application/pdf`.

Reply:
cool, `output binary` works in Studio using 4.3.0. And when ordered `output application/pdf with binary`, it also worked. Updated to the latter.